substra/sdk/schemas.py

```diff
@@ -147,14 +147,31 @@ def is_many(self):
         return self.paths and len(self.paths) > 0
 
     @pydantic.model_validator(mode="before")
-    def exclusive_paths(cls, values):  # noqa: N805
+    def exclusive_paths(cls, values: typing.Any) -> typing.Any:  # noqa: N805
         """Check that one and only one path(s) field is defined."""
         if "paths" in values and "path" in values:
             raise ValueError("'path' and 'paths' fields are exclusive.")
         if "paths" not in values and "path" not in values:
             raise ValueError("'path' or 'paths' field must be set.")
```

Comment:
Could they define both keys but with value `None`?

Reply:
I don't get what you're refering too. Could you provide an example plz ?

Reply:
It looks like we allow `None` as value on these fields, therefore if a user explicitely define only one of this key to `None` (e.g. `DataSampleSpec(path=None, data_manager_keys=[])`, the check `"paths" not in values and "path" not in values` will return False (as the key is present, but the value is `None`).

Reply:
Ok I see. I added some tests, and it appears that this case is dealts with in the resolve path, as `None` will raise a `TypeError`. This is not an explicit solution but I added the tests to ensure that this behavior is taken into account. What do you think ?

Reply:
I don't know if `TypeError` is the clearest for the user, but maybe it is worth opening its own PR to review how we manage this edge case?

Reply:
Yep True... I rm the tests, and +1 to deal with this in a dedicated PR